src/python/pants/backend/python/target_types.py

```diff
@@ -878,6 +878,43 @@ class PythonTestsXdistConcurrencyField(IntField):
     )
 
 
+class PythonTestsBatchCompatibilityTagField(StringField):
+    alias = "batch_compatibility_tag"
+    help = softwrap(
+        """
+        An arbitrary value used to mark the test files belonging to this target as valid for
+        batched execution.
+
+        It's _sometimes_ safe to run multiple `python_test`s within a single `pytest` process,
+        and doing so can give significant wins by allowing reuse of expensive test setup /
+        teardown logic. To opt into this behavior, set this field to an arbitrary non-empty
+        string on all the `python_test` targets that are safe/compatible to run in the same
+        process.
+
+        If this field is left unset on a target, the target is assumed to be incompatible with
+        all others and will run in a dedicated `pytest` process.
+
+        If this field is set on a target, and its value is different from the value on some
+        other `python_test`, then the two targets are explicitly incompatible and are guaranteed
+        to not run in the same `pytest` process.
+
+        If this field is set on a target, and its value is the same as the value on some other
+        `python_test`, then the two targets are explicitly compatible and _may_ run in the same
+        `pytest` process. Compatible tests may not end up in the same `pytest` batch if:
+
+            * There are "too many" compatible tests in a partition, as determined by the \
+                `[test].batch_size` config parameter, or
+            * Compatible tests have some incompatibility in Pants metadata (i.e. different \
+                `resolve`s or `extra_env_vars`).
```

Comment:
Thoughts on adding logs in this situation?

Reply:
Not opposed - in our repo we don't (yet) use the Pants features that cause this sort of incompatibility so I'm not sure how to judge the noisiness vs. usefulness

Reply:
Maybe debug level to start